packages/commonwealth/client/scripts/views/components/Header/LoginSelector.tsx

```diff
@@ -90,7 +90,7 @@ export const LoginSelector = () => {
     );
   }
 
-  if (!profileLoadComplete && NewProfilesController.Instance.allLoaded()) {
+  if (!profileLoadComplete) {
     setProfileLoadComplete(true);
   }
```

Comment:
1. is this even necessary here? If I removed this `if` statement, login selector seems to work fine? If not necessary, this code below could be removed
```
  useEffect(() => {
    NewProfilesController.Instance.isFetched.on('redraw', () => {
      setProfileLoadComplete(true);
    });
  }, []);
  ```
2. If this is necessary though, I know this is not your code, but setting state should not be in render (it should be in useEffect)

Reply:
Hm, `NewProfilesController.Instance.isFetched.on(` actually sets up a listener, so the state will update on the listener callback. 

TBH, I don't understand why `if (!profileLoadComplete) {` was added originally but it seem to sometimes break the login on metamask in my tests. We can do more digging here I suppose, will do more and update...

Reply:
I know that it sets up a listener but the question is if we need this listener here? As I said, I removed it along with the setProfileLoadComplete and I did not see any errors. User Avatar has loaded properly. Please test it around and remove if everuthing works. We want eventually get rid of  `NewProfilesController.Instance.isFetched` anyways